src/clj/witan/send/send.clj

```diff
@@ -334,80 +341,93 @@
+                                (-> (medley/map-vals round stats)
+                                    (assoc :need (name need) :calendar-year year)))
+                              (:total-in-send-by-need output))) send-output (range initial-projection-year 3000))
+               (map (apply juxt columns))
+               (concat [(map name columns)])
+               (csv/write-csv writer))))
+      (with-open [writer (io/writer (io/file "target/output-setting.csv"))]
+        (let [columns [:calendar-year :setting :mean :std-dev :iqr :min :low-ci :q1 :median :q3 :high-ci :max]]
+          (->> (mapcat (fn [output year]
+                         (map (fn [[setting stats]]
+                                (-> (medley/map-vals round stats)
+                                    (assoc :setting (name setting) :calendar-year year)))
+                              (:total-in-send-by-setting output))) send-output (range initial-projection-year 3000))
+               (map (apply juxt columns))
+               (concat [(map name columns)])
+               (csv/write-csv writer))))
+      (with-open [writer (io/writer (io/file "target/output-count.csv"))]
+        (let [columns [:calendar-year :mean :std-dev :iqr :min :low-ci :q1 :median :q3 :high-ci :max]]
+          (->> (map (fn [stats year]
+                      (-> (medley/map-vals round stats)
+                          (assoc :calendar-year year)))
+                    (map :total-in-send send-output) (range initial-projection-year 3000))
+               (map (apply juxt columns))
+               (concat [(map name columns)])
+               (csv/write-csv writer))))
+      (with-open [writer (io/writer (io/file "target/output-cost.csv"))]
+        (let [columns [:calendar-year :mean :std-dev :iqr :min :low-ci :q1 :median :q3 :high-ci :max]]
+          (->> (map (fn [stats year]
+                      (-> (medley/map-vals round stats)
+                          (assoc :calendar-year year)))
+                    (map :total-cost send-output) (range initial-projection-year 3000))
+               (map (apply juxt columns))
+               (concat [(map name columns)])
+               (csv/write-csv writer))))
+      (with-open [writer (io/writer (io/file "target/output-ay-group.csv"))]
+        (let [columns [:calendar-year :ay-group :mean :std-dev :iqr :min :low-ci :q1 :median :q3 :high-ci :max]]
+          (->> (mapcat (fn [output year]
+                         (map (fn [[ay-group stats]]
+                                (-> (medley/map-vals round stats)
+                                    (assoc :ay-group ay-group :calendar-year year)))
+                              (:total-in-send-by-ay-group output))) send-output (range initial-projection-year 3000))
+               (map (apply juxt columns))
+               (concat [(map name columns)])
+               (csv/write-csv writer))))
+      (run! #(ch/sankey-transitions transitions-data % valid-settings) years)))
```

Comment:
just to be sure I understand, this line and lines around 360 are the only changes in this block?

Reply:
Yes all of the the code output and the new plot file output have been put in an `if` so when running tests it's not creating additional files